src/java.base/share/classes/java/lang/reflect/code/op/CoreOps.java

```diff
@@ -1661,9 +1661,8 @@ static List<Value> operands(Value array, Value index, Value v) {
         }
 
         static TypeElement resultType(Value array, Value v) {
-            ArrayType arrayType = (ArrayType) array.type();
-            if (!arrayType.isArray()) {
-                throw new IllegalArgumentException("Type is not an array type: " + arrayType);
+            if (!(array.type() instanceof ArrayType arrayType)) {
```

Comment:
Note how the old code was buggy - as we did the test _after_ casting to `ArrayType`